src/components/SQForm/useFormButton.js

```diff
@@ -3,26 +3,59 @@ import {useFormikContext} from 'formik';
 import {useDebouncedCallback} from 'use-debounce';
 import {hasUpdated} from '../../utils';
 
-export function useFormButton(isDisabled, shouldRequireFieldUpdates, onClick) {
-  const {values, initialValues, isValid, ...rest} = useFormikContext();
+export const BUTTON_TYPES = {
+  SUBMIT: 'submit',
+  RESET: 'reset'
+};
+
+export function useFormButton({
+  isDisabled = false,
+  shouldRequireFieldUpdates = false,
+  onClick,
+  buttonType
+}) {
+  const {values, initialValues, isValid, dirty, ...rest} = useFormikContext();
   const hasFormBeenUpdated = hasUpdated(initialValues, values);
```

Comment:
[Question] Does anyone know if `hasFormBeenUpdated` is ever different from `dirty`? The Formik docs for `dirty` say that it is calculated by deep comparing `initialValues` and `values` but it looks like this is also the implementation of `hasUpdated` so not sure if these two ways of calculating basically the same thing are different/necessary.

Reply:
> [Question] Does anyone know if `hasFormBeenUpdated` is ever different from `dirty`? The [Formik docs](https://formik.org/docs/api/formik#dirty-boolean) for `dirty` say that it is calculated by deep comparing `initialValues` and `values` but it looks like this is also the implementation of `hasUpdated` so not sure if these two ways of calculating basically the same thing are different/necessary.

Forgot I wrote this util function awhile back. 
I'm pretty sure we can delete that `util` function which would also allow us to remove the fast-deep-equal dependency. I don't believe it's used anywhere else.